webapi/src/main/scala/org/knora/webapi/slice/search/api/SearchEndpoints.scala

```diff
@@ -67,49 +162,183 @@ object SearchEndpoints {
+  ): Task[(RenderedResponse, MediaType)] = for {
+    resourceClass <- ZIO.foreach(limitByResourceClass.map(_.value))(iriConverter.asSmartIri)
+    searchResult <-
+      searchResponderV2.searchResourcesByLabelV2(query, offset.value, project, resourceClass, opts.schema, user)
+    response <- renderer.render(searchResult, opts)
+  } yield response
+
+  def searchResourcesByLabelCountV2(
+    query: String,
+    opts: FormatOptions,
+    project: Option[ProjectIri],
+    limitByResourceClass: Option[InputIri]
+  ): Task[(RenderedResponse, MediaType)] = for {
+    resourceClass <- ZIO.foreach(limitByResourceClass.map(_.value))(iriConverter.asSmartIri)
+    searchResult <-
+      searchResponderV2.searchResourcesByLabelCountV2(query, project, resourceClass)
+    response <- renderer.render(searchResult, opts)
+  } yield response
+
+  def gravsearch(query: String, opts: FormatOptions, user: UserADM): Task[(RenderedResponse, MediaType)] = for {
+    searchResult <- searchResponderV2.gravsearchV2(query, opts.schemaRendering, user)
+    response     <- renderer.render(searchResult, opts)
+  } yield response
+
+  def gravsearchCount(query: String, opts: FormatOptions, user: UserADM): Task[(RenderedResponse, MediaType)] = for {
+    searchResult <- searchResponderV2.gravsearchCountV2(query, user)
+    response     <- renderer.render(searchResult, opts)
+  } yield response
+
+  def fullTextSearch(
+    query: RenderedResponse,
+    opts: FormatOptions,
+    offset: Offset,
+    project: Option[ProjectIri],
+    resourceClass: Option[InputIri],
+    standoffClass: Option[InputIri],
+    returnFiles: Boolean,
+    user: UserADM
+  ): Task[(RenderedResponse, MediaType)] = for {
+    resourceClass <- ZIO.foreach(resourceClass.map(_.value))(iriConverter.asSmartIri)
+    standoffClass <- ZIO.foreach(standoffClass.map(_.value))(iriConverter.asSmartIri)
+    searchResult <- searchResponderV2.fulltextSearchV2(
+                      query,
+                      offset.value,
+                      project,
+                      resourceClass,
+                      standoffClass,
+                      returnFiles,
+                      opts.schemaRendering,
+                      user
+                    )
+    response <- renderer.render(searchResult, opts)
+  } yield response
+
+  def fullTextSearchCount(
+    query: RenderedResponse,
+    opts: FormatOptions,
+    project: Option[ProjectIri],
+    resourceClass: Option[InputIri],
+    standoffClass: Option[InputIri]
+  ): zio.Task[
+    (_root_.org.knora.webapi.slice.common.api.KnoraResponseRenderer.RenderedResponse, _root_.sttp.model.MediaType)
+  ] = for {
+    resourceClass <- ZIO.foreach(resourceClass.map(_.value))(iriConverter.asSmartIri)
+    standoffClass <- ZIO.foreach(standoffClass.map(_.value))(iriConverter.asSmartIri)
+    searchResult  <- searchResponderV2.fulltextSearchCountV2(query, project, resourceClass, standoffClass)
+    response      <- renderer.render(searchResult, opts)
```

Comment:
```suggestion
final case class SearchRestService(
  responder: SearchResponderV2,
  renderer: KnoraResponseRenderer,
  iriConverter: IriConverter
) {

  def searchResourcesByLabelV2(
    query: String,
    opts: FormatOptions,
    offset: Offset,
    project: Option[ProjectIri],
    limitByResourceClass: Option[InputIri],
    user: UserADM
  ): Task[(RenderedResponse, MediaType)] = for {
    resourceClass <- ZIO.foreach(limitByResourceClass.map(_.value))(iriConverter.asSmartIri)
    searchResult <-
      responder.searchResourcesByLabelV2(query, offset.value, project, resourceClass, opts.schema, user)
    response <- renderer.render(searchResult, opts)
  } yield response

  def searchResourcesByLabelCountV2(
    query: String,
    opts: FormatOptions,
    project: Option[ProjectIri],
    limitByResourceClass: Option[InputIri]
  ): Task[(RenderedResponse, MediaType)] = for {
    resourceClass <- ZIO.foreach(limitByResourceClass.map(_.value))(iriConverter.asSmartIri)
    searchResult <-
      responder.searchResourcesByLabelCountV2(query, project, resourceClass)
    response <- renderer.render(searchResult, opts)
  } yield response

  def gravsearch(query: String, opts: FormatOptions, user: UserADM): Task[(RenderedResponse, MediaType)] = for {
    searchResult <- responder.gravsearchV2(query, opts.schemaRendering, user)
    response     <- renderer.render(searchResult, opts)
  } yield response

  def gravsearchCount(query: String, opts: FormatOptions, user: UserADM): Task[(RenderedResponse, MediaType)] = for {
    searchResult <- responder.gravsearchCountV2(query, user)
    response     <- renderer.render(searchResult, opts)
  } yield response

  def fullTextSearch(
    query: RenderedResponse,
    opts: FormatOptions,
    offset: Offset,
    project: Option[ProjectIri],
    resourceClass: Option[InputIri],
    standoffClass: Option[InputIri],
    returnFiles: Boolean,
    user: UserADM
  ): Task[(RenderedResponse, MediaType)] = for {
    resourceClass <- ZIO.foreach(resourceClass.map(_.value))(iriConverter.asSmartIri)
    standoffClass <- ZIO.foreach(standoffClass.map(_.value))(iriConverter.asSmartIri)
    searchResult <- responder.fulltextSearchV2(
                      query,
                      offset.value,
                      project,
                      resourceClass,
                      standoffClass,
                      returnFiles,
                      opts.schemaRendering,
                      user
                    )
    response <- renderer.render(searchResult, opts)
  } yield response

  def fullTextSearchCount(
    query: RenderedResponse,
    opts: FormatOptions,
    project: Option[ProjectIri],
    resourceClass: Option[InputIri],
    standoffClass: Option[InputIri]
  ): zio.Task[
    (_root_.org.knora.webapi.slice.common.api.KnoraResponseRenderer.RenderedResponse, _root_.sttp.model.MediaType)
  ] = for {
    resourceClass <- ZIO.foreach(resourceClass.map(_.value))(iriConverter.asSmartIri)
    standoffClass <- ZIO.foreach(standoffClass.map(_.value))(iriConverter.asSmartIri)
    searchResult  <- responder.fulltextSearchCountV2(query, project, resourceClass, standoffClass)
    response      <- renderer.render(searchResult, opts)
```

Reply:
I appreciate the effort in renaming. However, I will not integrate it using GitHub as I do not have compiler support in case you have missed something.